spec/02-integration/20-wasm/01-admin-api_spec.lua

```diff
@@ -428,7 +427,7 @@ describe("wasm admin API [#" .. strategy .. "]", function()
       fcs = {
         assert(bp.filter_chains:insert({
           filters = {
-            { name = "tests", config = ngx.null, enabled = true },
```

Comment:
Do we have tests for cases with YAML/JSON inputs, in DB-less or hybrid mode?

(This one change prompted me to ask because I recall we have conversion steps which add `ngx.null`s in missing fields in certain places, so seeing this being removed made me think of that and wonder if some behavior changed for `config = ngx.null` that might need to be handled)

Reply:
:+1: good to bring this up. So [spec/02-integration/20-wasm/09-filter-meta_spec.lua](https://github.com/Kong/kong/blob/1e4a488dea4e121b97390f9cbcaceb80bc1fa836/spec/02-integration/20-wasm/09-filter-meta_spec.lua) has some dbless test coverage, as does [spec/02-integration/20-wasm/05-cache-invalidation_spec.lua](https://github.com/Kong/kong/blob/1e4a488dea4e121b97390f9cbcaceb80bc1fa836/spec/02-integration/20-wasm/05-cache-invalidation_spec.lua).

Overall though I would rate the current dbless test coverage as "not great." I've done a lot of work to increase this test coverage in the last couple days, but I'm waiting until after this `config`/`json_config`/`raw_config` business is settled to submit a PR for it (and including it here would increase the size of this PR by 2/3x).

I'll make sure we have some explicit JSON `null` test cases in the test update PR. _Technically speaking_ `null` is another weird edge case like JSON strings, but it's even more tricky because it requires that we carefully disambiguate `null`/`nil` when an entity is deserialized from an API request. My 2c is that it's not worth trying to disambiguate them (at least right now): if a Wasm filter can accept `null` as a valid configuration, it's reasonable for it to also accept an empty (zero-length) configuration.

Reply:
> Technically speaking null is another weird edge case like JSON strings

For sure. I'm not concerned about "accepting JSON configs which consist of a single JSON `null`" — as far as this PR is goes, I just want to make sure that the Gateway doesn't trip up and behave differently on `config = nil` versus `config = ngx.null`, since IIRC we have code paths which automatically convert one into the other, ahead of the entities getting to the `db` calls.

If the easy way to ensure we'll have no surprises with this is to _not_ accept JSON configs which consist of a single JSON `null` (i.e. `config = ngx.null` is treated as `config = nil`), I'm fine with that.